app/controllers/samples_controller.rb

```diff
@@ -36,7 +36,8 @@ class SamplesController < ApplicationController
   before_action :authenticate_user!, except: TOKEN_AUTH_ACTIONS
   before_action :authenticate_user_from_token!, only: TOKEN_AUTH_ACTIONS
 
-  before_action :admin_required, only: [:reupload_source, :resync_prod_data_to_staging, :kickoff_pipeline, :retry_pipeline, :pipeline_runs]
+  before_action :admin_required, only: [:reupload_source, :resync_prod_data_to_staging, :kickoff_pipeline, :retry_pipeline,
```

Comment:
As discussed offline, lets replace this check by a feature check... 
You can declare a `before_action` that checks for you feature and configure it for your methods.
You can check using: `current_user.allowed_feature_list.include?("pipeline-viz")` and you can add the feature using `user.add_allowed_feature("pipeline-viz")`.
This way you can keep the rest of the code.